src/vs/workbench/contrib/terminalContrib/accessibility/browser/textAreaSyncAddon.ts

```diff
@@ -17,30 +19,47 @@ export interface ITextAreaData {
 export class TextAreaSyncAddon extends Disposable implements ITerminalAddon {
 	private _terminal: Terminal | undefined;
 	private _onCursorMoveListener = this._register(new MutableDisposable());
+	private _onDidFocusListener = this._register(new MutableDisposable());
+	private _onKeyListener = this._register(new MutableDisposable());
 	activate(terminal: Terminal): void {
 		this._terminal = terminal;
 		if (this._accessibilityService.isScreenReaderOptimized()) {
-			this._onCursorMoveListener.value = this._terminal.onCursorMove(() => this._refreshTextArea());
+			this._setListeners();
 		}
 	}
 
 	constructor(
 		private readonly _capabilities: ITerminalCapabilityStore,
+		private readonly _onDidFocus: Event<ITerminalInstance>,
 		@IAccessibilityService private readonly _accessibilityService: IAccessibilityService,
 		@ITerminalLogService private readonly _logService: ITerminalLogService
 	) {
 		super();
 		this._register(this._accessibilityService.onDidChangeScreenReaderOptimized(() => {
-			if (this._accessibilityService.isScreenReaderOptimized() && this._terminal) {
+			if (this._accessibilityService.isScreenReaderOptimized()) {
 				this._refreshTextArea();
-				this._onCursorMoveListener.value = this._terminal.onCursorMove(() => this._refreshTextArea());
+				this._setListeners();
 			} else {
 				this._onCursorMoveListener.clear();
+				this._onDidFocusListener.clear();
+				this._onKeyListener.clear();
```

Comment:
If you're working with all the listeners together you could use a `MutableDisposable<DisposableStore>` so there's a single member prop